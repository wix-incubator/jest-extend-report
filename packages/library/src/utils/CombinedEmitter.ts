import type { Emitter, ReadonlyEmitter } from '../types';
import { SerialEmitter } from './SerialEmitter';

export class CombinedEmitter<Event extends { type: string }> implements ReadonlyEmitter<Event> {
  private readonly emitters = new WeakSet<Emitter<Event>>();
  private readonly rootEmitter = new SerialEmitter<Event>('combined');

  add(emitter: Emitter<Event>): this {
    if (!this.emitters.has(emitter)) {
      this.emitters.add(emitter);
      emitter.on('*', (event: Event) => /* re-emit */ this.rootEmitter.emit(event));
    }

    return this;
  }

  on(type: Event['type'] | '*', listener: (event: Event) => void): this {
    this.rootEmitter.on(type, listener);
    return this;
  }

  once(type: Event['type'] | '*', listener: (event: Event) => void): this {
    this.rootEmitter.once(type, listener);
    return this;
  }

  off(type: Event['type'] | '*', listener: (event: Event) => void): this {
    this.rootEmitter.off(type, listener);
    return this;
  }
}
