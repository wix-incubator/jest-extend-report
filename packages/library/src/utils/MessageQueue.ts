import { SerialEmitter } from './SerialEmitter';

export type SendMethod<Message> = (message: Message) => Promise<unknown>;

export type MessageQueueEvent<Message> =
  | { type: 'error'; error: Error }
  | { type: 'drain' }
  | { type: 'next'; message: Message };

export class MessageQueue<Message> extends SerialEmitter<MessageQueueEvent<Message>> {
  private _idle: Promise<unknown> = Promise.resolve();
  private _queue: Message[] = [];
  constructor(protected readonly send: SendMethod<Message>) {
    super();
  }

  public enqueue(message: Message): this {
    this._queue.push(message);
    this._idle = this._idle.then(this._unshift, this._unshift);
    return this;
  }

  public flush(): Promise<unknown> {
    return this._idle;
  }

  private _unshift = () => {
    const message = this._queue.shift();
    /* istanbul ignore next */
    if (!message) {
      this.emit({ type: 'drain' });
      return;
    }

    this.emit({ type: 'next', message });
    return this.send(message).catch((error) => this.emit({ type: 'error', error }));
  };
}
