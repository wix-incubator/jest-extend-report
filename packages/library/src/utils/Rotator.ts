export class Rotator<T> {
  #items: T[] = [];
  #offset = 0;

  get size() {
    return this.items.length;
  }

  get offset() {
    return this.#offset;
  }

  get items(): readonly T[] {
    return [...this.#items.slice(this.offset), ...this.#items.slice(0, this.offset)];
  }

  find(predicate: (item: T) => boolean): T | undefined {
    let i = this.size;
    while (i--) {
      const item = this.next()!;
      if (predicate(item)) {
        return item;
      }
    }

    return;
  }

  reset(): this {
    this.#offset = 0;
    return this;
  }

  add(...items: T[]): this {
    this.#items.push(...items);
    return this;
  }

  next(): T | undefined {
    const item = this.#items[this.offset];
    this.#offset = (this.offset + 1) % this.items.length;
    return item;
  }
}
