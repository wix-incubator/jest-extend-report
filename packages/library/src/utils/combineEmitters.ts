import type { ReadonlyEmitter } from '../types';
import { SerialEmitter } from './SerialEmitter';

export function combineEmitters<A extends { type: string }, B extends { type: string }>(
  a: ReadonlyEmitter<A>,
  b: ReadonlyEmitter<B>,
): ReadonlyEmitter<A | B> {
  const emitter = new SerialEmitter<A | B>();
  a.on('*', (event) => emitter.emit(event));
  b.on('*', (event) => emitter.emit(event));
  return emitter;
}
