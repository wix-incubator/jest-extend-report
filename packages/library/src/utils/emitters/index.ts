export * from './AggregatedEmitter';
export * from './SerialAsyncEmitter';
export * from './SerialSyncEmitter';
export * from './SemiAsyncEmitter';
