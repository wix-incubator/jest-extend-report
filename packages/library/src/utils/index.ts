export { SerialEmitter } from './SerialEmitter';
export { MessageQueue } from './MessageQueue';
export { CombinedEmitter } from './CombinedEmitter';
export { getVersion } from './getVersion';
export { logger, optimizeForLogger } from './logger';
