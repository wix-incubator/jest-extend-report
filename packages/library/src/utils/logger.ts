import { traceEventStream, wrapLogger } from 'bunyamin';
import { createLogger } from 'bunyan';
import { noop } from './noop';

export const logger = create();

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const optimizeForLogger: <F>(f: F) => F = isEnabled() ? (f) => f : ((() => noop) as any);

function create() {
  return wrapLogger({
    logger: (isEnabled() ? createBunyanImpl : createBunyanNoop)(),
  });
}

function isEnabled(): boolean {
  return process.env.JEST_METADATA_DEBUG === 'true';
}

function createBunyanImpl() {
  const label = process.env.JEST_WORKER_ID ? `Worker ${process.env.JEST_WORKER_ID}` : 'Main';
  const suffix = process.env.JEST_WORKER_ID ? `-${process.env.JEST_WORKER_ID}` : '';

  const bunyan = createLogger({
    name: `Jest Metadata (${label})`,
    streams: [
      {
        level: 'trace' as const,
        stream: traceEventStream({
          filePath: `jest-metadata.${process.pid}${suffix}.log`,
          threadGroups: [
            { id: 'ipc-server', displayName: 'IPC Server' },
            { id: 'ipc-client', displayName: 'IPC Client' },
            { id: 'emitter-core', displayName: 'Emitter (core)' },
            { id: 'emitter-set', displayName: 'Emitter (set)' },
            { id: 'emitter-combined', displayName: 'Emitter (combined)' },
            { id: 'metadata', displayName: 'Metadata' },
            { id: 'reporter', displayName: 'Reporter' },
          ],
        }),
      },
    ],
  });

  return bunyan;
}

function createBunyanNoop() {
  return {
    trace: noop,
    debug: noop,
    info: noop,
    warn: noop,
    error: noop,
    fatal: noop,
  };
}
