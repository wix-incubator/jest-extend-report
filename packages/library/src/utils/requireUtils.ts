import path from 'path';
import type { TestEnvironment as JsdomTestEnvironment } from 'jest-environment-jsdom';

export function requireJestEnvironmentNode() {
  return require('jest-environment-node');
}

export function requireJestEnvironmentJsDom() {
  return requireViaChain('jest-environment-jsdom') as typeof JsdomTestEnvironment;
}

function requireViaChain(targetModuleName: string, chain?: string[]) {
  let cwd = process.cwd();
  if (chain) {
    for (const module of chain) {
      cwd = path.dirname(require.resolve(module + '/package.json', { paths: [cwd] }));
    }
  }
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const theModule = requireViaPath(targetModuleName, cwd);
  return theModule;
}

function requireViaPath(modulePath: string, viaPath: string) {
  return require(require.resolve(modulePath, { paths: [viaPath] }));
}
