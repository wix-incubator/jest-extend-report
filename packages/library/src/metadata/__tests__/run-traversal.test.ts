import fixtures from '@jest-metadata/fixtures';

import {
  AggregatedMetadataRegistry,
  Metadata,
  MetadataEventHandler,
  MetadataFactoryImpl,
  SetMetadataEventEmitter,
} from '..';

import { SerialSyncEmitter } from '../../utils';

describe('run metadata traversal:', () => {
  const lastFixtures = Object.values(fixtures).filter(([name]) => {
    return name.startsWith('29.x.x') && name.includes('-worker-N');
  });

  test.each(lastFixtures)(`fixtures/%s`, (_name: string, fixture: any[]) => {
    const emitter: SetMetadataEventEmitter = new SerialSyncEmitter('set');
    const metadataRegistry = new AggregatedMetadataRegistry();
    const metadataFactory = new MetadataFactoryImpl(metadataRegistry, emitter);
    const aggregatedResultMetadata = metadataFactory.createAggregatedResultMetadata();
    const eventHandler = new MetadataEventHandler({
      aggregatedResultMetadata,
      metadataRegistry,
    });

    for (const event of fixture) {
      eventHandler.handle(event);
    }

    const lastRun = aggregatedResultMetadata.lastTestResult!;
    if (!lastRun) {
      return;
    }

    const toId = (x: Metadata) => {
      const [a, b] = x.id.split(':');
      return b || a;
    };

    const toChain = (x: Metadata[]) => x.map(toId).join(' â†’ ');

    expect(toChain([...lastRun.allDescribeBlocks()])).toMatchSnapshot('allDescribeBlocks');
    expect(toChain([...lastRun.allTestEntries()])).toMatchSnapshot('allTestEntries');
    expect(toChain([...lastRun.allTestInvocations()])).toMatchSnapshot('allTestInvocations');
    expect(toChain([...lastRun.allInvocations()])).toMatchSnapshot('allInvocations');
  });
});
