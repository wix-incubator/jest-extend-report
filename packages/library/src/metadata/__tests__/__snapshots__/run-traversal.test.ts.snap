// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-cases: allDescribeBlocks 1`] = `"describe_0 → describe_1 → describe_2"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-cases: allInvocations 1`] = `"hook_0.0 → test_0.0.fn → hook_0.1 → test_1.0.fn → hook_0.2 → test_2.0.fn → hook_0.3 → test_3.0.fn"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-cases: allTestEntries 1`] = `"test_0 → test_1 → test_2 → test_3"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-cases: allTestInvocations 1`] = `"test_0.0 → test_1.0 → test_2.0 → test_3.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-generation: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-generation: allInvocations 1`] = `"hook_0.0 → test_0.0.fn → hook_0.1 → test_1.0.fn → hook_0.2 → test_0.1.fn"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-generation: allTestEntries 1`] = `"test_0 → test_1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-generation: allTestInvocations 1`] = `"test_0.0 → test_1.0 → test_0.1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-name: allDescribeBlocks 1`] = `"describe_0 → describe_1 → describe_2 → describe_3"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-name: allInvocations 1`] = `"hook_1.0 → hook_2.0 → hook_0.0 → test_0.0.fn → hook_0.1 → test_1.0.fn → hook_0.2 → test_1.1.fn → hook_3.0 → hook_4.0 → hook_0.3 → test_4.0.fn → hook_0.4 → test_5.0.fn → hook_0.5 → test_4.1.fn → hook_5.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-name: allTestEntries 1`] = `"test_0 → test_1 → test_2 → test_3 → test_4 → test_5"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/duplicate-name: allTestInvocations 1`] = `"test_0.0 → test_1.0 → test_1.1 → test_2.0 → test_3.0 → test_4.0 → test_5.0 → test_4.1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-afterAll-fail: allDescribeBlocks 1`] = `"describe_0 → describe_1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-afterAll-fail: allInvocations 1`] = `"hook_0.0 → test_0.0.fn → hook_2.0 → hook_1.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-afterAll-fail: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-afterAll-fail: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-beforeAll-fail: allDescribeBlocks 1`] = `"describe_0 → describe_1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-beforeAll-fail: allInvocations 1`] = `"hook_1.0 → test_1.0.fn"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-beforeAll-fail: allTestEntries 1`] = `"test_0 → test_1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-beforeAll-fail: allTestInvocations 1`] = `"test_0.0 → test_1.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-nesting: allDescribeBlocks 1`] = `"describe_0 → describe_1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-nesting: allInvocations 1`] = `"hook_1.0 → hook_0.0 → hook_2.0 → test_0.0.fn → hook_3.0 → hook_0.1 → hook_2.1 → test_1.0.fn → hook_3.1 → hook_4.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-nesting: allTestEntries 1`] = `"test_0 → test_1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/hook-nesting: allTestInvocations 1`] = `"test_0.0 → test_1.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-concurrent: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-concurrent: allInvocations 1`] = `"test_0.0.fn → test_1.0.fn → test_2.0.fn → test_3.0.fn → test_4.0.fn"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-concurrent: allTestEntries 1`] = `"test_0 → test_1 → test_2 → test_3 → test_4"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-concurrent: allTestInvocations 1`] = `"test_0.0 → test_1.0 → test_2.0 → test_3.0 → test_4.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-fail: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-fail: allInvocations 1`] = `"hook_0.0 → test_0.0.fn"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-fail: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-fail: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-pass: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-pass: allInvocations 1`] = `"hook_0.0 → test_0.0.fn"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-pass: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-pass: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-retry: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-retry: allInvocations 1`] = `"hook_1.0 → hook_0.0 → test_0.0.fn → hook_0.1 → test_0.1.fn → hook_2.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-retry: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-retry: allTestInvocations 1`] = `"test_0.0 → test_0.1"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-skip: allDescribeBlocks 1`] = `"describe_0 → describe_1 → describe_2"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-skip: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-skip: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-skip: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-todo: allDescribeBlocks 1`] = `"describe_0 → describe_1 → describe_2"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-todo: allInvocations 1`] = `"hook_1.0 → hook_2.0 → hook_3.0 → hook_4.0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-todo: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/env-worker-N/test-todo: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-cases: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-cases: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-cases: allTestEntries 1`] = `"test_0 → test_1 → test_2 → test_3"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-cases: allTestInvocations 1`] = `"test_0.0 → test_1.0 → test_2.0 → test_3.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-generation: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-generation: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-generation: allTestEntries 1`] = `"test_0 → test_1"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-generation: allTestInvocations 1`] = `"test_0.0 → test_1.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-name: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-name: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-name: allTestEntries 1`] = `"test_0 → test_1 → test_2 → test_3 → test_4 → test_5"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/duplicate-name: allTestInvocations 1`] = `"test_0.0 → test_1.0 → test_2.0 → test_3.0 → test_4.0 → test_5.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-afterAll-fail: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-afterAll-fail: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-afterAll-fail: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-afterAll-fail: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-beforeAll-fail: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-beforeAll-fail: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-beforeAll-fail: allTestEntries 1`] = `"test_0 → test_1"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-beforeAll-fail: allTestInvocations 1`] = `"test_0.0 → test_1.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-nesting: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-nesting: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-nesting: allTestEntries 1`] = `"test_0 → test_1"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/hook-nesting: allTestInvocations 1`] = `"test_0.0 → test_1.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-concurrent: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-concurrent: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-concurrent: allTestEntries 1`] = `"test_0 → test_1 → test_2 → test_3 → test_4"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-concurrent: allTestInvocations 1`] = `"test_0.0 → test_1.0 → test_2.0 → test_3.0 → test_4.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-fail: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-fail: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-fail: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-fail: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-pass: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-pass: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-pass: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-pass: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-retry: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-retry: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-retry: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-retry: allTestInvocations 1`] = `"test_0.0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-skip: allDescribeBlocks 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-skip: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-skip: allTestEntries 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-skip: allTestInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-todo: allDescribeBlocks 1`] = `"describe_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-todo: allInvocations 1`] = `""`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-todo: allTestEntries 1`] = `"test_0"`;

exports[`run metadata traversal: fixtures/29.x.x/no-env-worker-N/test-todo: allTestInvocations 1`] = `"test_0.0"`;
